core/src/com/unciv/logic/automation/BarbarianAutomation.kt

```diff
@@ -0,0 +1,224 @@
+package com.unciv.logic.automation
+
+import com.unciv.Constants
+import com.unciv.UncivGame
+import com.unciv.logic.battle.Battle
+import com.unciv.logic.battle.BattleDamage
+import com.unciv.logic.battle.MapUnitCombatant
+import com.unciv.logic.civilization.CivilizationInfo
+import com.unciv.logic.map.MapUnit
+import com.unciv.logic.map.PathsToTilesWithinTurn
+import com.unciv.logic.map.TileInfo
+import com.unciv.models.AttackableTile
+import com.unciv.models.UnitAction
+import com.unciv.models.ruleset.unit.UnitType
+import com.unciv.ui.worldscreen.unit.UnitActions
+
+class BarbarianAutomation(val civInfo: CivilizationInfo) {
+
+    private val battleHelper = BattleHelper()
+    private val battleDamage = BattleDamage()
+
+    fun automate() {
+        applyPromotions()
```

Comment:
Barbarians shouldn't ever have promotions though...?

Reply:
Oh I thought they could, ok I'll remove this